File: storm-json/src/test/java/st/orm/json/IntegrationConfig.java
package st.orm.json;

import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.transaction.annotation.EnableTransactionManagement;

@Configuration
@EntityScan
@ComponentScan("st.orm.json")
@EnableJpaRepositories
@EnableTransactionManagement
public class IntegrationConfig {
}

File: storm-kotlin/src/test/kotlin/st/orm/kt/IntegrationConfig.kt
package st.orm.kt

import org.springframework.boot.jdbc.DataSourceBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.datasource.DataSourceTransactionManager
import org.springframework.transaction.annotation.EnableTransactionManagement
import st.orm.kt.template.ORMTemplate
import javax.sql.DataSource

@Configuration
@EnableTransactionManagement
open class IntegrationConfig {

    @Bean
    open fun dataSource(): DataSource =
        DataSourceBuilder.create()
            .url("jdbc:h2:mem:testdb;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=false")
            .username("sa")
            .password("")
            .driverClassName("org.h2.Driver")
            .build()

    @Bean
    open fun transactionManager(dataSource: DataSource): DataSourceTransactionManager =
        DataSourceTransactionManager(dataSource)

    @Bean
    open fun ormTemplate(dataSource: DataSource): ORMTemplate =
        ORMTemplate.of(dataSource)
}